examples/lib/stories/camera_and_viewport/zoom.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:flame/gestures.dart';
+
+class ZoomGame extends BaseGame with ScrollDetector, ScaleDetector {
+  final Vector2 viewportResolution;
+  late SpriteComponent flame;
+
+  Vector2? lastDelta;
+
+  ZoomGame({
+    required this.viewportResolution,
+  });
+
+  @override
+  Future<void> onLoad() async {
+    final flameSprite = await loadSprite('flame.png');
+
+    viewport = FixedResolutionViewport(viewportResolution);
+
+    final flameSize = Vector2(149, 211);
+    add(
+      flame = SpriteComponent(
+        sprite: flameSprite,
+        size: flameSize,
+      )..anchor = Anchor.center,
+    );
+    camera.followComponent(
+      flame,
+      relativeOffset: Anchor.center.toVector2(),
+    );
+  }
+
+  static const zoomPerScrollUnit = 0.001;
+  @override
+  void onScroll(PointerScrollInfo event) {
+    camera.zoom += event.scrollDelta.game.y * zoomPerScrollUnit;
+  }
+
+  @override
+  void onScaleUpdate(ScaleUpdateInfo info) {
+    final scale = lastDelta;
+    if (scale != null) {
+      final delta = info.scale.game - scale;
+      camera.zoom += delta.y;
```

Comment:
you should multiply/divide the current zoom by the delta (abs), not add, right?

Reply:
I don't think so, this particularly operation is working as expected.

This code is responsible for the effect you see on the mobile example on the video on the PR description.

Reply:
I find it weird that the change in scale doesn't consider the current scale, hence my proposal. Also your implementation doesn't guard against negative values, which is inherent with an additive solution vs a multiplicative one (but can be solved in other ways)

Reply:
I may be missing something, but I am considering the current scale to calculate the new one.

I don't really think that this can get negative, because the nature of the gesture, and in my tests, this worked fine. I could try changing to your idea, but I am having a hard time picturing it without seeing some code.